Updates to readme (#2)
Guidewire/cda-client: README.md

# Cloud Data Access Client

The Cloud Data Access Client (CDA Client) is a utility to download table data from an Amazon S3 bucket that has been populated by Guidewire's Cloud Data Access (CDA). The utility reads the .parquet files generated by CDA for each table, and can do one of the following:

- Convert them into human readable .csv files

> - Transfer them as .parquet files

Reviewer: The parquet files that CDA client puts out are not identical to the files that CDA writes, so I think this is a bit misleading.
```suggestion
- Aggregate them into fewer `.parquet` files
```

> - Load the data into a common database platform: SQL Server, Oracle, PostgreSQL. 

Reviewer: ```suggestion
- Load the data into a database: SQL Server, Oracle, PostgreSQL. 
```

The example code can be used to jump-start development of a custom implementation that can store CDA data in other storage formats (e.g., RDBMS Tables, Hadoop/Hive, JSON Files).

> Learn more about CDA [here](https://https://docs.guidewire.com/cloud/cda/banff/index.html).

Reviewer: ```suggestion
Learn more about CDA [here](https://docs.guidewire.com/cloud/cda/banff/index.html).
```

- - - 
# Overview of CDA Client
<details>
<summary>Click to expand</summary>

> When converting to .csv files, the utility provides the schema for each table in a ``schema.yaml`` file, and can be configured to output the resultant files into a designated local filesystem location or another Amazon S3 bucket. When writing to a database, the data can be loaded in its raw format, with each activity recorded from the source system database, or it can be merged into tables that more closely resemble the source system database.

Reviewer: ```suggestion
When converting CDA output to `.csv` files, the utility provides the schema for each table in a `schema.yaml` file, and can be configured to put these files into a local filesystem location or another Amazon S3 bucket. When writing to a database, the data can be loaded in "raw" format, with each insert/update/delete recorded from the source system database, or it can be merged into tables that more closely resemble the source system database.
```

The utility also resumes downloading from the point which it last read up to when rerun, so that new data in the source bucket can be read efficiently and incrementally over multiple runs. 
### Code Structure

- Written in Scala

> - Main program starts in gw.cda.api.CloudDataAccessClient

Reviewer: ```suggestion
- Main program starts in `gw.cda.api.CloudDataAccessClient`
```

1. **Configure S3 authentication**: The utility requires AWS credentials to access the source S3 bucket that contains the table data (and optionally to write to a destination S3 bucket). These must be exported in the command line before running the program. For example, use this command after replacing `<secret key>`, `<access key>`, and `<region name>` with values from a user's credentials file located in `~/.aws/credentials`:
~~~~
export AWS_SECRET_ACCESS_KEY=<secret key> AWS_ACCESS_KEY_ID=<access key> AWS_REGION=<region name>
~~~~

   - If you are using awscli with a credentials file and profiles, you will need to use the environment var `AWS_PROFILE`, instead of setting the keys directly. For example:  
~~~~
export AWS_PROFILE=myProfile
~~~~
* More information can be found here for setting up [AWS Credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html)

2. Download the sample configuration file from the Git repository folder `/src/test/resources/sample_config.yaml` and save under a new name such as `config.yaml`.
3. Configure the `config.yaml` file.
4. Run the utility by executing the jar from the command line with one of these commands:

>   <ul><li>If you are running the CDA Client for the first time (without a <tt>savepoints.json</tt> file from a previous run) or have a large amount of data in the S3 bucket, both reading and writing can take a substantial amount of time depending on your machine. By default, the Java runtime environment [allocates a maximum of 1/4 of the computer's memory](https://docs.oracle.com/javase/8/docs/technotes/guides/vm/gc-ergonomics.html). It may be necessary to increase the memory available to the application for larger amounts of data. For example, run the client with an increased maximum memory allocation of 8 GB ("8g") with this command:

Reviewer: ```suggestion
  <ul><li>If you are running the CDA Client for the first time (without a `savepoints.json` file from a previous run) or have a large amount of data in the S3 bucket, both reading and writing can take a substantial amount of time depending on your machine. By default, the Java runtime environment [allocates a maximum of 1/4 of the computer's memory](https://docs.oracle.com/javase/8/docs/technotes/guides/vm/gc-ergonomics.html). It may be necessary to increase the memory available to the application for larger amounts of data. For example, run the client with an increased maximum memory allocation of 8 GB ("8g") with this command:
```

`

> In the local filesystem, the client jar and config.yaml file exist in the current directory, along with a directory in which to contain the .csv outputs:

Reviewer: ```suggestion
In the local filesystem, the client jar and `config.yaml` file are in the current directory, along with a directory in which to contain the `.csv` outputs:
```
